falcon/routing/compiled.py

```diff
@@ -112,16 +112,20 @@ def line(text, indent_offset=0):
         level_indent = indent
         found_simple = False
 
-        # NOTE(kgriffs): Sort static nodes before var nodes so that
-        # none of them get masked. False sorts before True.
-        nodes = sorted(nodes, key=lambda node: node.is_var)
+        # NOTE(kgriffs & philiptzou): Sort nodes in this sequence:
+        # static nodes(0), complex var nodes(1) and simple var nodes(2).
+        # so that none of them get masked.
```

Comment:
@yohanboniface The registering order is not important since we'll sort the nodes inside our code. That's here.